### PR TITLE
Set minimum C++ version to be C++11

### PR DESCRIPTION
**Pull request by seze4251**
Hey All,

I just pulled down GoogleTest on my MacOS and tried to build it locally.  I ran into some build errors with CMake because clang was defaulting to a C++ version less than C++11.  I edited the CMake files in GoogleTest, and GoogleMock to enforce a minimum version of C++11, which will allow GoogleTest to work out-of-the-box on MacOS.

Thanks,
Seth

### CODE REVIEW

**Comment by seze4251** [comment]
@googlebot I signed it!

File: .gitignore [hunk 1]
```diff
@@ -24,6 +24,9 @@ Win32-Release/
 x64-Debug/
 x64-Release/
 
+# Eclipse
```
Comment (derekmauro) [inline .gitignore:27]: Please revert the changes to this file. This really belong in a global file <https://docs.github.com/en/github/using-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer> otherwise this file will just be a collection of IDE specific directories.

File: googlemock/CMakeLists.txt [hunk 2]
```diff
@@ -78,6 +78,11 @@ set(gmock_build_include_dirs
   "${gtest_SOURCE_DIR}")
 include_directories(${gmock_build_include_dirs})
 
+# Set minimum C++ version to C++11
+set(CMAKE_CXX_STANDARD 11)
```
Comment (derekmauro) [inline googlemock/CMakeLists.txt:82]: If we accept this change, how would I build GoogleTest if I wanted C++20 support?

File: .gitignore (hunk 1 above)
Comment (seze4251) [reply .gitignore]: reverted the .gitignore.  Putting the IDE stuff in a global git ignore is a really good idea.  My company just shoves them in our local repos, so I was used to that.  I will use your approach going forward.  Thanks!

File: googlemock/CMakeLists.txt (hunk 2 above)
Comment (seze4251) [reply googlemock/CMakeLists.txt:82]: The other pull request that just got merged is better.  I will close this.
